pyfr/readers/cgns.py

```diff
@@ -0,0 +1,433 @@
+# -*- coding: utf-8 -*-
+
+from collections import defaultdict
+from ctypes import (POINTER, RTLD_GLOBAL, create_string_buffer,
+                    c_char_p, c_int, c_void_p)
+import re
+
+import numpy as np
+
+from pyfr.ctypesutil import load_library
+from pyfr.readers import BaseMeshConverter, BaseReader
+from pyfr.readers.nodemaps import CGNSNodeMaps
+from pyfr.util import setenv
+
+
+class CGNSWrappers(object):
+    def __init__(self):
+        # Load CGNS
+        try:
```

Comment:
I found that CGNS 3.3 fix this issue. Comparing to CGNS 3.3-rc1, they changed cmake option related with HDF5 and now it can automatically load hdf5 as well. I try to call _load_hdf5 (I prefer this name) only when we got OSError.
Regarding CGNS version, CGNS 3.2+ supports cubic (P3) elements and CGNS 3.2.2+ supports quartic (P4) elements.